applications/asset_tracker_v2/src/cloud/lwm2m_integration/lwm2m_integration.c

```diff
@@ -178,15 +181,26 @@ static void rd_client_event(struct lwm2m_ctx *client, enum lwm2m_rd_client_event
 		break;
 	case LWM2M_RD_CLIENT_EVENT_NETWORK_ERROR:
 		LOG_ERR("LWM2M_RD_CLIENT_EVENT_NETWORK_ERROR");
-		cloud_wrap_evt.type = CLOUD_WRAP_EVT_DISCONNECTED;
+		cloud_wrap_evt.type = CLOUD_WRAP_EVT_ERROR;
 		notify = true;
-		state = DISCONNECTED;
 		break;
 	default:
 		LOG_ERR("Unknown event: %d", client_event);
 		break;
 	}
 
+	/* If a LwM2M failure has occurred, we explicitly stop the engine before the cloud module
+	 * is notified with the CLOUD_WRAP_EVT_DISCONNECTED event. This is to clear up any
+	 * LwM2M engine state to ensure that we are able to perform a clean restart of the engine.
+	 */
+	if (notify && cloud_wrap_evt.type == CLOUD_WRAP_EVT_DISCONNECTED) {
+		err = cloud_wrap_disconnect();
```

Comment:
`err` is only used within this if-block, so why not define it here instead of at the top of the function?